frontend/src/machines/billingMachine.ts

```diff
@@ -1,3 +1,137 @@
-import { createMachine } from "xstate";
+import { createMachine, assign } from "xstate";
 
-export const billingMachine = createMachine({});
+type Billing = {
+  id: string;
+  title: string;
+  amount: number;
+};
+
+interface Context {
+  billings: Billing[];
+  isEditingForm: boolean;
+  isCreatingForm: boolean;
+}
+
+const billingMachine = createMachine<Context>({
+  id: "billing",
+  initial: "idle",
+  context: {
+    billings: [],
+    isCreatingForm: false,
+    isEditingForm: false,
+  },
+  states: {
+    idle: {
+      on: {
+        FETCHINGBILLINGS: {
+          target: "gettingBilling",
+        },
+      },
+    },
+    gettingBilling: {
+      on: {
+        FETCHINGBILLINGSSUCCES: {
+          target: "gettingBillingOK",
+        },
+        FETCHINGBILLINGSERROR: {
+          target: "gettingBillingError",
+        },
+      },
+    },
+    gettingBillingOK: {
+      on: {
+        ACTIVATEBILLINGFORM: {
+          target: "billingFormIdle",
+        },
+      },
+    },
+    gettingBillingError: {
+      on: {
+        REFETCHINGBILLINGS: {
+          target: "gettingBilling",
+        },
+      },
+    },
+    billingFormIdle: {
+      always: [
+        {
+          cond: "creatingForm",
+          target: "#firstStep",
+        },
+        {
+          target: "gettingBillingDetail",
+        },
+      ],
+    },
+    gettingBillingDetail: {
+      on: {
+        FETCHINGBILLINGDETAILSUCCES: {
+          target: "#firstStep",
+        },
+        FETCHINGBILLINGDETAILERROR: {
+          target: "gettingBillingDetailError",
+        },
+      },
+    },
+    gettingBillingDetailError: {
+      on: {
+        REFTECHBILLINGDETAIL: {
+          target: "gettingBillingDetail",
+        },
+      },
+    },
+    billingFormReady: {
+      initial: "firstStep",
+      states: {
+        firstStep: {
+          id: "firstStep",
+          on: {
+            NEXTSTEP: {
+              target: "secondStep",
+            },
+          },
```

Comment:
Ini kayaknya belum ada action buat update value formnya ya ?

Reply:
Will be handle on the next MR yak, karena dari context nya emang masih belum lengkap hehe